### PR TITLE
chore: limit concurrent operations while migrating workspaces

### PR DESCRIPTION
**Pull request by DemonHa**
## Fix: Limit Concurrent Writes During Migration

Previously, there was no limit on concurrent writes to IndexedDB during the migration process.  
For users with a large number of workspaces, this could lead to performance issues and potential instability.

### Changes

- Added a concurrency limit for writes to **IndexedDB** during migration.
- Added a concurrency limit for document insertions into the workspace store.
- Improved overall migration stability for users with many workspaces and documents.

These changes ensure the migration process is more predictable, resource-efficient, and less likely to overwhelm the browser.

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset. 
- [ ] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Medium Risk**
> Touches the localStorage→IndexedDB migration path and changes execution ordering/concurrency for writes and data transforms, which could affect migration completeness/performance if limits or queueing have edge cases. Risk is mitigated by adding comprehensive `createLimiter` tests and keeping logic changes localized.
> 
> **Overview**
> Limits concurrency during the localStorage→IndexedDB workspace migration by throttling **IndexedDB `setItem` writes** and **workspace/document transformation work** via `createLimiter` (new caps like `MAX_CONCURRENT_DB_WRITES` and `MAX_CONCURRENT_DATA_TRANSFORMATIONS`).
> 
> Moves `json-magic` to use the shared helpers `createLimiter`, updates `createLimiter` to use the shared `Queue` implementation, and adds a more complete test suite while removing the old `json-magic`-local test. A changeset bumps `@scalar/json-magic`, `@scalar/oas-utils`, and `@scalar/helpers` as patch releases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3cd4a10c8c5f358899d6454f949ea7ee1d5120d2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/helpers/src/general/create-limiter.test.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+describe('create-limiter', () => {
+  beforeEach(() => {
+    vi.useFakeTimers()
+  })
+  afterEach(() => {
+    vi.useRealTimers()
+  })
+
+  it('run in order and no more than the specified number of concurrent requests', async () => {
+    const limiter = createLimiter(2)
```
Comment (geoffgscott) [inline packages/helpers/src/general/create-limiter.test.ts:14]: Feels like limiter should just accept an array of promises and an options argument. Then it can just we awaited rather than always managing an instance and having to wrap in Promise.all anyways.

```
function limiter<T>(Promise<T>[], options: { concurrency?: number }): Promise<T[]>

**Comment by hanspagel** [comment]
I think @DemonHa needs this urgently, so let's merge it
